crates/node-api/src/engine/traits.rs

```diff
@@ -2,12 +2,38 @@ use crate::{validate_version_specific_fields, AttributesValidationError, EngineA
 use reth_primitives::{
     revm::config::revm_spec_by_timestamp_after_merge,
     revm_primitives::{BlobExcessGasAndPrice, BlockEnv, CfgEnv, SpecId},
-    Address, ChainSpec, Header, B256, U256,
+    Address, ChainSpec, Header, SealedBlock, B256, U256,
 };
-use reth_rpc_types::engine::{
-    OptimismPayloadAttributes, PayloadAttributes as EthPayloadAttributes, PayloadId, Withdrawal,
+use reth_rpc_types::{
+    engine::{
+        ExecutionPayloadEnvelopeV2, ExecutionPayloadEnvelopeV3, OptimismPayloadAttributes,
+        PayloadAttributes as EthPayloadAttributes, PayloadId, Withdrawal,
+    },
+    ExecutionPayloadV1,
 };
 
+/// Represents a built payload type that contains a built [SealedBlock] and can be converted into
+/// engine API execution payloads.
+pub trait BuiltPayload: Send + Sync + std::fmt::Debug {
+    /// Initializes the payload with the given initial block.
+    fn new(id: PayloadId, block: SealedBlock, fees: U256) -> Self;
```

Comment:
hmm, I think this is fine now, but this effectively restricts how the built payload looks like.

I see that we're using this as the fallback for empty payloads, which is a default.

let's track this separately, so we can improve this a bit.
perhaps it should be the job's responsibility to create an empty payload